docs/source/provisioners.rst

```diff
@@ -0,0 +1,95 @@
+Provisioners
+============
+
+Molecule uses provisioners to bring up Ansible ready hosts to operate on.
+Currently, Molecule supports two provisioners: Vagrant and Docker.
+
+The provisioner can set when using ``init`` command or through the
+``molecule.yml`` file.
+
+
+Docker Provisioner
+------------------
+The docker provisioner is compatible with any image
+that has python installed. Molecule will automatically install
+python for images with the yum or apt-get package tools. A new
+docker image will be built with the prefix molecule_local to separate it
+from the other images on your system.
+
+Below is an example of a ``molecule.yml`` file using two containers ``foo-01`` and
+``foo-02``. ``foo-01`` is running the latest image of ubuntu and ``foo-02`` is running
+the latest image of ubuntu from a custom registry.
+
+The available params for docker containers are:
+
+* ``name`` - name of the container
+* ``ansible_groups`` - groups the container belongs to in Ansible
+* ``image`` - name of the image
+* ``image_version`` - version of the image
+* ``registry`` - **(OPTIONAL)** the registry to obtain the image
+
+Docker Example
+--------------
+
+.. code-block:: yaml
+
+    ---
+    docker:
+      containers:
+        - name: foo-01
+          ansible_groups:
+          - group1
+            image: ubuntu
+            image_version: latest
+          - name: foo-02
+            ansible_groups:
+              - group2
+            image: ubuntu
+            image_version: latest
+            registry: testhost:5323
+
+
+
+Vagrant Provisioner
+-------------------
+
+The vagrant provisioner performs in a similar manner to the docker provisioner.
+Except for using virtual machines instead of containers. Each instance of a vagrantbox
+are defined inside of an instance with similar options to docker. The provisioner is
+set to vagrant by default if the ``--docker`` flag is not passed when ``molecule init`` is run.
+
+The available parameters for vagrant instances are:
+
+* ``name`` - name of the vagrant box
+* ``ansible_groups`` - groups the instance belongs to in ansible
+* ``interfaces`` - network inferfaces (see ``usage``)
+* ``options`` - Vagrant options supported by Molecule
+* ``raw_config_args`` - Vagrant options unsupported by Molecule
+
+Vagrant Instance Example
+------------------------
+This is an example of a set of vagrant instance - for information on specifying the platform/
+provider, see :ref:`providers`.
+
+.. code-block:: yaml
+
+    ---
+      instances:
```

Comment:
looks like we need to dedent this 2 spaces.  `---` doesn't line up with `instances`.